src/withImmer.ts

```diff
@@ -7,14 +7,25 @@ const cache1 = new WeakMap();
 const memo1 = <T>(create: () => T, dep1: object): T =>
   (cache1.has(dep1) ? cache1 : cache1.set(dep1, create())).get(dep1);
 
+export function withImmer<Value, Args extends unknown[], Result>(
+  anAtom: WritableAtom<Value, Args, Result>,
+): WritableAtom<
+  Value,
+  Args extends [Value | infer OtherValue]
+    ? [
+        | Value
+        | ((draft: Draft<Value>) => void)
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        | Exclude<OtherValue, (...args: any[]) => unknown>,
```

Comment:
Does `never` work here?
```suggestion
        | Exclude<OtherValue, (...args: never[]) => unknown>,
```

Reply:
Yeah, it does actually?? No idea how this make sense semantically, but I have been doing things wrong all this time then :)))